File: Java/src/main/java/org/codecop/lcdnumbers/DigitsSplitter.java
package org.codecop.lcdnumbers;

import java.util.List;
import java.util.Objects;
import java.util.stream.Collectors;

/**
 * Splits into digits and converts digits to LCD digits.
 */
public class DigitsSplitter {

    private final NumeralSystem numeralSystem;
    private final DigitFactory digitFactory;

    public DigitsSplitter(NumeralSystem numeralSystem, DigitFactory digitFactory) {
        Objects.requireNonNull(numeralSystem);
        Objects.requireNonNull(digitFactory);

        this.numeralSystem = numeralSystem;
        this.digitFactory = digitFactory;
    }

    public List<Digit> convert(int number) {
        List<Integer> numeralDigits = numeralSystem.digitsOf(number);

        return numeralDigits.stream(). //
                map(digitFactory::create). //
                collect(Collectors.toList());
    }
}

File: Csharp-MsDependencyInjection/LcdNumbers/DigitsSplitter.cs
using System.Collections.Generic;
using System.Linq;

namespace Org.Codecop.Lcdnumbers
{
    /// <summary>Splits into digits and converts digits to LCD digits.</summary>
    public class DigitsSplitter : IDigitsSplitter
    {
        private readonly INumeralSystem numeralSystem;
        private readonly IDigitFactory digitFactory;

        public DigitsSplitter(INumeralSystem numeralSystem, IDigitFactory digitFactory)
        {
          this.numeralSystem = numeralSystem;
          this.digitFactory = digitFactory;
        }

        public virtual IList<Digit> Convert(int number)
        {
            IList<int> numeralDigits = numeralSystem.DigitsOf(number);
            return numeralDigits.Select(d => digitFactory.Create(d)).ToList();
        }
    }
}
